eslint-rules/src/rules/use-correct-component-overrides.ts

```diff
@@ -0,0 +1,136 @@
+import { Selectors } from '@angular-eslint/utils';
+import { ASTUtils, TSESLint, TSESTree } from '@typescript-eslint/experimental-utils';
+import * as fs from 'fs';
+import * as path from 'path';
+
+import { normalizePath } from '../helpers';
+
+export const useCorrectComponentOverridesRule: TSESLint.RuleModule<string, []> = {
+  meta: {
+    messages: {
+      shouldPointToBasicFile: 'Override should point to basic component file.',
+      testOverrideTemplateMissing:
+        'Did not find template override ({{ expectedTemplate }}) this spec is supposed to test.',
+      testOverrideTSMissing: 'Did not find TS import ({{ expectedTS }}) this spec is supposed to test.',
+      pointToBasicFile: 'Replace with basic component file.',
+    },
+    type: 'problem',
+    schema: [],
+    hasSuggestions: true,
+  },
+  create(context) {
+    const fileName = path.basename(normalizePath(context.getFilename()));
+    const fileBaseName = fileName.replace(/component.*/, 'component');
+    if (/\.component\.(?!(spec|stories))([a-z]+\.)?ts$/.test(fileName)) {
+      return {
+        [`${Selectors.COMPONENT_CLASS_DECORATOR} ObjectExpression > Property[key.name="templateUrl"] > Literal`](
+          node: TSESTree.Literal
+        ) {
+          const expected = `./${fileBaseName}.html`;
+          if (node.value !== expected) {
+            context.report({
+              node,
+              messageId: 'shouldPointToBasicFile',
+              suggest: [{ fix: fixer => fixer.replaceText(node, `'${expected}'`), messageId: 'pointToBasicFile' }],
+            });
+          }
+        },
+
+        [`${Selectors.COMPONENT_CLASS_DECORATOR} ObjectExpression > Property[key.name="styleUrls"] > ArrayExpression`](
+          node: TSESTree.ArrayExpression
+        ) {
+          const expected = `./${fileBaseName}.scss`;
+          if (!node.elements.some(el => el.type === TSESTree.AST_NODE_TYPES.Literal && el.value === expected)) {
+            const suggest = [];
+            if (node.elements.length <= 1) {
+              suggest.push({
+                fix: fixer => fixer.replaceText(node, `['${expected}']`),
+                messageId: 'pointToBasicFile',
+              });
+            }
+
+            context.report({
+              node,
+              messageId: 'shouldPointToBasicFile',
+              suggest,
+            });
+          }
+        },
+      };
+    } else if (/\.component\.[a-z]+\.spec\.ts$/.test(fileName)) {
+      function classify(base: string): string {
```

Comment:
I feel like we have too many implementations of these similar string manipulation functions in the PWA. I think it would be nice to at least put this method in the `helpers.ts` so people coming after won't add yet another implementation. 
Btw, why not use `strings.classify` from `@angular-devkit/core` like we do for schematics?

Reply:
Good idea using `strings` 👍